packages/visualizations/src/components/Chart/legend.ts

```diff
@@ -67,3 +67,67 @@ export default function buildLegend(
     };
     return legend;
 }
+
+function buildLegendLabels(
+    index: number,
+    options: ChartOptions,
+    chartConfig: ChartConfiguration
+): string {
+    const text = options?.legend?.labels?.text;
+    if (text) {
+        return assureMaxLength(text(index), LEGEND_MAX_LENGTH);
+    }
+    return `${chartConfig.data.labels?.[index]}`;
+}
+
+export function buildCustomLegend({
+    chart,
+    options,
+    chartConfig,
+}: {
+    chart: Chart;
+    options: ChartOptions;
+    chartConfig: ChartConfiguration;
+}) {
+    const { series } = options;
+    return {
+        type: 'category' as const,
+        position: defaultValue(options?.legend?.position, 'bottom'),
+        items: chartConfig.data.datasets[0].data.map((_data, i) => ({
+            color: series[0].backgroundColor?.[i],
+            borderDashed: false,
+            label: buildLegendLabels(i, options, chartConfig),
+            onClick: (index: number) => {
+                if (chart) {
+                    chart.toggleDataVisibility(index);
```

Comment:
Using `toggleDataVisibility` here causes a crash in `pieDataLabelsPlugin.ts` as we do not check there if data is visible and we try to draw lines on hidden data : 

On line 22 in `pieDataLabelsPlugin.ts` I think we should wrap the drawing inside a check 
`if (chart.getDataVisibility(i)) { ... }`
This way there is no crash and the hover seems to work again !